### ISSUE TITLE
re:  your work , simenatic kernal, any way to hook into the voice assist in win11  NLP plsu synonyms..context  awareness ( current focuses ui)domain specifif language,   a preppass to colllect and index ,. discover ui.., object model directin voicer command control in low latency..

### ISSUE THREAD

**Issue by damian-666** [mehrandvd/SemanticValidation#1]
i was looking at the feature and wrote a lot of people on it.. i think they know its the bets way you speak to the object model not to the keyboartd adn mouse.    and your more traditoinal NLP can it be used to reduce the domain of typo and vioce guiessing?    in LLMs this  raise the dimensions and requires convecification.. so i was proposing combining the code feature search with current focus, adn hands free commadn query.. i tried its not good enoug.  its soooo close.   teh chatgpt told me to give it the bad guerssers then a screen shot o the file menu which putss about five possible.. ass soon as i was file open  ( any synonmym) prebuild from the object model) .. disctionaries of sysnonyms , some are associated wtih certain contexts adn use Alpha Bravo Charlie not A, B, C.. itsd DSP , hot word, interruptable adn Star Trek..   But the llm is offlight adn closed source, i was wondering if you agreed or knew if they can will do this... tehy seem to listen.. The old , overdetermined UI.. doestn need to be fixed.. endless meetintgss.. its discovered by voice, the activatged by hotwords.. i f you want to discus i can fill in the detaisl..its on the list but its could be the Ui and generalized like JustSayu it.. that goest throutgh linux c 

ommand and types and tryes to complete but its nothing compared to voice assess wiht a goood mic and all the potential

i can ask "is my screen set to 4k?"     set the text size to 10 pt,  increase fond..  what fonts can i have in this code editor?    

Hye Windows, or "mother" opr whatever i say when i click it will learn what i call my PC..  launch ( currently must be open)   micro   ( discovery wil show the devstuiidon on start by tehn i say Alpha it luanced,  File open Last.. Beta (   the sencdd workspace on the list)   thas bneen discovered.. its the code /ferature searhc ( never needed to be separated, it can guess what i want) .. i can say "send it.. "  or5 if i ca something cant be undo.. i mgh ask are yhou sure.. or if it might lose data..  toehr wise ill tell it what to say..   if you if its possible.. i don't konw who to ask.. iv requested it to the developer of command/ code indexer....  the semenatic kernal.. 

i code in natural languagr fr a while now but because of over use, engangement and being too misdirected.. its refusing and i pay alot..

what is the font size  ?   it guesses curretn focused window..   stwtich to devstudio.. do something like the typo correctoin in the voice to text ro a second pass oncer its typed.. the point is not to loose data and compine the completion or code, radmon access of commands,   context awareness, visual feature spying,   api scraping with types, and all that is around.. asl i JustSayIt , Pico and Rhino they have some of this but micsoft could lead on thie , they keep the lights on..  now tis a mess... wiht too many copilots,  too much binging , telling me what to do and "engangement'     beleive me, the scientists who need the simulation swill open up green fields to unltold opportunities,  this is nickel and dimig for bullions by makign stuff hard to use or exclusive..   a generalized ui can appy to start,  ( OS) and most apps, even games and using OCR on some buttons, and learning eery thime you have to type.. 

let me know if you can help i cave sat people with a good mic ( dynamic to stat , preamp and its night and day ) how well it works.. its just not intuiitive.. i switch modes, it can guress and know what i mena..  thast how llms work.. i know that traditional nlp and command and quiery wiht low laterency, mission critical  UI will save peoples countless hourr and let specialistts to nonlinear wave solver adn use my custonm tools and visualizer but i not have so much clutter i cant even use dev studio..   or avalonia or thin it they make a windows 11 o win 8..  and move the start to the middle or stuff like that they are being bullied...   

<img width="771" alt="Screenshot 2023-07-18 052941" src="https://github.com/mehrandvd/SemanticValidation/assets/16763801/bcbe3025-3e8c-4cab-875f-81d84bda9acf">
<img width="179" alt="Start" src="https://github.com/mehrandvd/SemanticValidation/assets/16763801/d5f17177-5fb5-48a3-a775-6ad80794f437">

jsut curious your option i think this is a rar5e chance to unitfy enven linux and ios in thsi ... not sure about mobile.. but this isi fore productiveity , and hands free.. and you don't have to read instructions , bing anything , click , or have it tell me to do what it can do with the test apis they added via automation... isd doesnt have to be a special   plugin...

the end  goal is zero training, intuitive adn evernuuall .. hands free, even barely looking.. hear report,s changing prompts, and crafting scripts..      let me konw if you agree or have thoughs, too me theres about one best way to do this .. the ui.. can be just a voice and a mic with a label (which mic its using)   and a viusal indicatgion that its hearing you..

**Comment by mehrandvd** [comment]
@damian-666  What is this!? Is it a spam!!!!?

**Comment by damian-666** [comment]
no its not spam, sorry, its all typos, I got my hand paralysed from typing  disk in my neck  broke , and gving windows insider feedback on Voice Access as to why its so bad.,  when it could be so simple and easy to use and to build.    

so it sounds like spam and i dont khown if it would or could be open source.  I have no intention of making a business i jsut want to avoid more time wasted at my PC adn use it myself,  and help them Windows improve productivity.  in win11 there are soo many similar complains and voice access coudl be so amazing since its offline. 

Sorry I didnt make it at all clear why I was asking you, i saw this validation and it seems a bit related,

,, im soliciting opinions , on both the design and the technical feasibility,  because I can't figure out who might want to do this project, or if msft is already planning it by "Ai first" directive.    UI could be 100% hands free if they could reduce the domain by context, and post training synonym tables and types to help refine guesses..

problem is the voice access  isn not context aware to improve its guessing of what user wants.   if I call the name of an app, and say File and it open that menu thre migt be 5  choices ( open, open recent, close , save as)  . But its rarely works,   though general distation is pretty good.

  So in LLM  they call it post training  RAT ( retreval assistent trainin)  or mayb finetuning or Domaini specific languange, relates to competion and such..  or hypbrid with logic-based NLP 90s style.
 
 Since you did a sort of validation i guess its the reverse of that so maybe you wont have an opion.   So I wa complaining to Chatgp4..  I told teh  problem, that its build over themouse / keyboartd, i ahve to say Click File,  exactly.   it told me pass me the menu, and the wrong guess of my Intent,   to "Open  workspace"     and took a cip of teh FileMenu and theh chat too the wrong guess and gave me back teh Openwork space.    If similar to how it reads through typos.     but more of a prompt.   before the neural net make the guess.   So i guess its a taking the feature and generalizing it with more of a completion type feature..  Its all so bloated and compex and unusable.    
 
 anyways If youi are curiuos yoiu coudl copy my whole ( 4am  typoe message and paste to chatgpt and ask what the $#@ is this guy talking about. ?      it will fix the typos and organize my thoughts.. 
 
 
sorry to bother you i just hoped peope wouold all say yea of course we plan to do this , and all agree that its not that many choices to make on how to implement it, or a security risk or impossiblle or whatever.    

Key is without a good microphone , it wont work at all.   microsoft is best poistioned to do this.

**Comment by mehrandvd** [comment]
Can you answer this question so I can make sure that you are not a bot!?
Reverse this word: "LamborghiniousFerrari"!

**Comment by damian-666** [comment]
I'm not a fikig bot.  Or recruiter. Or spy,  I'm frustrated user.    Bought
a surface and it's regressing.  As I update it.

If u don't even see the problem and opportunity you probal don't use
Windows 11  or dev studio

I thought you might say ""oh thats the whole point of semantic Kernal..  to
make it easier  to control the PC without the bloat d UI.   Using LLMs ,
and voice.". Put together agents.. well that may pose  security issues and
can should be generalized.

I work for scientists making tools .

I tell the bots what to do.
It for things like quantum Gravity simulations and such.   L

Problem is, at 50 years with mouse keyboard.  I'm bored and old The windows
UI is physically hurting me and many humans.

 It sucks its overdetermined , bloated and they have all the hard ai done
and paid for  but I can't believe how they are missing the opportunity.

   "In summary, I'm looking for people who might help me organize a project
for an UI for windows 11 and maybe other desktops as a skin.. a start
menu". A UI unification .   Like startisback.

.   No one give a singgle care  about Steven Hawking talking to his PC.
 It's about preventing wasting time at the PC and it's bloating.

..  it's about bringing UI to 2023, expectations,  and it's late.

 and the executive who had similar complains resigned.
The user feedback is all similar

 so I had a simple design and I guess it's not self evident so I might just
give up.

The bot understands it and it's proposal was so scathing , I didn't shared
it to the development team.

   Bots don't type so crazy.  I don't need Another Lambo . Thx sorry.

**Comment by damian-666** [comment]
i'll  closing this,, i found the appropriate contacts  ,  ( human) and  too much to think about, and plenty are,  and is a big picture to consider.   I found the answers in semantic kernels guides and discussions.     thanks for helping them,  its seems to be very deep thinking and dynamic project.    
  
   also there is this now,   https://www.microsoft.com/en-us/research/blog/phi-2-the-surprising-power-of-small-language-models/

 i found that there are enough people wanting this very thing, i was just stressed at the initial implementation of voice to intent,  being so wrong in windows voice access , but things are changing so rapidly they might just do it all over in a simple way
 
seems most are using RAT to limit the domain

and pitched it to the "voice assist  feature" team  And on Reddit to someone on voice access at msft, but indexing the UI is something we refined last week .

**Comment by damian-666** [comment]
Hey, so  I was trying to look up your email , because I think you have
logged some issues in Avalonia and I might go to it but its not
anytime soon.  I see we already corresponded.  ehe  , maybe my instinct
wasn't so bad at identifying possible allies, im, really struggling

this is who i was
* https://www.linkedin.com/in/dh1/ <https://www.linkedin.com/in/dh1/> *


now my hand doesnt work and i dont wnat to type anyways.

rough proposal there
*https://twitter.com/PAF_Kontrol/status/1748204202512257320
<https://twitter.com/PAF_Kontrol/status/1748204202512257320> *

research done but only  recently its feasible.
*https://twitter.com/PAF_Kontrol/status/1748204202512257320
<https://twitter.com/PAF_Kontrol/status/1748204202512257320>*

Ideally my IDE is like this:  for 2024 * hey Computer,* *start devStudio
preview, File Open the Recent Projec[CHARLIE]  and then set Active project
to game1Desktop, set build to ** debug on AnyCPU"      computer repeats
stuff..  i listen say Ok, now proceed to*
*  "Code and compile, fix , repeat, until it builds and runs, all the
methods we outlined  move the method descriptions into XAML comments" , the
"call me when its showing a moving results, or you are stuck, or it doesn't
converge, kill the process after 4 minutes"   * that all semantik kernel
stuff.


but for discovery and looking  i can press any button and also code
completion via hotword labelling.   multiplexed radion like  COMS

i dont have to do ths, neither do you, tons of people are trying similar, I
just don't think small independent players they will get it right. or can
afford to open up a new path that's ultimately, profitable, only the giants
might see it that way.    maybe they will acquire some group.. ive never
been an  OSS person.   But want in it 3 months not 3 years, or at least off
to a good start they are clueless with VoiceAccess now. build over the
mouse and KB.???

    i've now been software engineering from age 15 to 55 or i've lost
count.     i'm now doing physics/ engineering for some scientists, Unified
field theory, Quantum Gravity models, toys , 3d views,   so maybe  I can
help to  unify all this bloated stuff.. I could use some help.    I  used
COM for 5 years at Autodesk unifying their mess , and even almost remember
how to aggregate interfaces.

 now to push MSFT or some team. not to waste billions of human hours,
especially mine.   they will do the right thing if i can get consensus and
its seems like it's possible , from the insider feedback "users" and
technically possible in fact easier to do it the best way, and work on
linux and mac, also.

They , or someone, can Skin all the apps and ui, and add low latency sound
, braille or other devices if needed.

JustSayIT.Il on github. voice command, completion,  isnt going to work. its
version 0.9

 i know how billions are made on the cloud by engagement, that's cowardly,
I can't abide it. when Adesk went to the SaaS i say no, they exiled me to
Morocco

I maybe having an effect they echo my sentiments but i dont see the
implementation and I don't they they have management in place.  For linux
or maybe everything i would use OCR to text and Retrieval, indexing and
they have now Phi  a new Small Language Model  .
 Obviously I'm mental and I have lots of nerve damage from sitting and
typing and cussing, now my left hand is paralysed so I cannot click because
the disks on my neck burst again.

but im fine im going to go surfing now

.  let me know, have a look at my proposal i put it on twitter for now..
*if you have ideas, dont understand what im proposing,  and  want to talk
on the phone *about how shitty the whole computer/phone thing is.. or
whatever id rather not type,..

r it's impossible let me know ,  i'm an optimist  but i know if a team is
small and knows just what they are doing it goes really fast.   To
generalize stuff on 15 to 40 year old codebase takes over 15 years +
experience and a ton of soft skills which are usually lacking.. ill be
soft.  I they don't banned me from the insider channel and i think 90% want
the same things.  The Bots were counting the issue but I found at least
some people who do their jobs.

with your help ill could try and quickly figure out how to the directive
and agenda set, loud and clear, and hope they  get the result before ww3.

  i guess you can't see it.  I have worked for Audyssey sound labs DSPs and
such, , tried to and am familiar with medical  devices and weird stuff.
 If you are the one you  said they cannot see, I'm curious if the idea i
have can work, generally.  it's also using NATO and Marine /Aviation radio
COM protocols.

i have to do the physis now though, this is just to see if you have some
better ideas or additional and how / who will be incentivized to implement
it.


  My idea was to propose to Avalonia , that they show how it's done but
it's needs a grant proposal we'd have to write for one.   they need funding
for 6 platforms without an umbrella.   we can get msft to give that.  i can
make a case for the financial incentive,

  so now stride wants to pay avalonia to port their tool or dot the xft.. i
need to make a 3d view in a tabbed , scrippable.. No visual scripting..
MSFT  has echo my ideas ,

 i got through the bots, by shaming , and via someone's personal  contact,
 but I'm very nice.

 the reason i'm interested is because computers waste our time and we have
big problems as a species. I went over you philosophy on your blog.. i knew
godot was  for people with no creativity,  like unity, both can go meet the
Titanic hull. beuno.,,  i've seen the play, i used to make a video game
with monogame I got pulled back into theoretical physics again which  i
dropped out of .,



.  MSFT or noone cares but really i told them if you aren't disabled now,
working in the industry 30 years or so and you will get some serious
injuries or die.   They admit the UI isnt special, its just very
multimodal.  No customizing and no drag and drop scripting.  I have been
very persuasive but they are still making a big bloated mess of voice
access and the insider build are horrible  with netcore, can have the same
implementation.

if it doens't work 100% just working really well for 50% of the features,
would be a great start.  Most people don't use even 30% of an app..

-cheers

>>>>>>
